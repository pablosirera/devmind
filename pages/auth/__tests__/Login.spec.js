import { shallowMount } from '@vue/test-utils'
import LoginView from '@/pages/auth/login.vue'

describe('Auth -> Login View', () => {
  const build = () => {
    const wrapper = shallowMount(LoginView)

    return {
      wrapper,
    }
  }
  test('should be render', () => {
    const { wrapper } = build()
    expect(wrapper.vm).toBeTruthy()
  })

  test('should match snapshot', () => {
    const { wrapper } = build()
    expect(wrapper.element).toMatchSnapshot()
  })

  test('should click on social login button', () => {
    const { wrapper } = build()
    const socialButton = wrapper.findComponent({ name: 'SocialLoginButton' })
    socialButton.trigger('click')
  })
})
