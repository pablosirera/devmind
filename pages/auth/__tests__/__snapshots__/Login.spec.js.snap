// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth -> Login View should match snapshot 1`] = `
<div
  class="h-full overflow-hidden"
>
  <div
    class="flex justify-end p-4"
  >
    <colorswitch-stub />
  </div>
   
  <h1
    class="text-center text-3xl font-bold"
  >
    Devmind
  </h1>
   
  <div
    class="login-body mt-12 pt-56 h-full flex flex-col items-center bg-light dark:bg-blue-main-800"
  >
    <socialloginbutton-stub
      text="Iniciar sesión con google"
      type="google"
    />
     
    <p>
      iniciar sesión con google
    </p>
  </div>
</div>
`;
